homeassistant/components/prowl/config_flow.py

```diff
@@ -0,0 +1,105 @@
+"""The config flow for the Prowl component."""
+
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Mapping
+from functools import partial
+import logging
+from typing import Any
+
+import prowlpy
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_API_KEY, CONF_NAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ProwlConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the Prowl component."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.api_key = ""
+        self.name = "Prowl"
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle user configuration."""
+        errors = {}
+
+        if user_input and user_input[CONF_API_KEY]:
+            self.api_key = user_input[CONF_API_KEY]
+            if user_input[CONF_NAME] and len(user_input[CONF_NAME]) > 0:
+                self.name = user_input[CONF_NAME]
+
+            errors = await self._validate_api_key(self.api_key)
+            if not errors:
+                return self.async_create_entry(
+                    title=self.name,
+                    data={
+                        CONF_API_KEY: self.api_key,
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): str,
+                    vol.Required(CONF_NAME): str,
+                }
+            ),
```

Comment:
Wrap in `add_suggested_values_to_schema` so the user doesn't need to re-fill the form if there are errors.
Maybe something like this which will also allow removing the extra logic checking the name:
```py
data_schema=self.add_suggested_values_to_schema(
    vol.Schema(
        {
            vol.Required(CONF_API_KEY): str,
            vol.Required(CONF_NAME): str,
        }
    ),
    user_input or {CONF_NAME: "Prowl"},
)
```